### PR TITLE
ABCI Request and Response types with ToJSON and FromJSON instances compliant with Tendermint JSON RPC

### PR DESCRIPTION
**Pull request by charlescrain**

### CODE REVIEW

**Comment by charlescrain** [comment]
FYI the brittany formatter changed some of the files, LMK if its not a style you like. 
https://github.com/f-o-a-m/hs-abci/pull/70/files#diff-645530c37356187833b470c48cde8e54R107

File: hs-abci-types/src/Network/ABCI/Types/Messages/FieldTypes.hs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
```
Comment (martyall) [inline hs-abci-types/src/Network/ABCI/Types/Messages/FieldTypes.hs]: we should put this in the `package.yaml` file

**Review by martyall — CHANGES_REQUESTED** [review]

**Comment by martyall** [comment]
How are you invoking this formatter? I don't see any kind of config file or `make` command.

**Comment by charlescrain** [comment]
> How are you invoking this formatter? I don't see any kind of config file or `make` command.

It's the default formatter in hie-vscode

**Comment by charlescrain** [comment]
I went through and fixed the formatting. I also wrote custom parsers where necessary. The parser for HexString now handles, null and parses it as an empty HexString (i.e. `0x`). Since the HexString packaged is defined for use within this package it seemed like a good idea as it saves writing a bunch of custom parsers. Let me know if this doesn't work and I'll dive in and do the custom parsers.

Also, this is not complete, still getting a parser error. when calling tendermint.

**Comment by charlescrain** [comment]
Found another problem with our encoding, we need to use `Base64` strings instead.

**Review by martyall — APPROVED** [review]

**Pull request merged**